File: P101VB/Form1.vb
Public Class Form1

    Dim total As Integer                ' Variavel guarda total
    Dim descontoEscalao As Integer      ' guarda valor desconto escalao
    Dim descontoSocio As Integer        ' guarda valor desconto Socio
    Dim agravamento As Integer = 12     ' guarda valor agravamento
    Dim descontoDistrito As Integer     ' guarda valor desconto distrito
    Dim descontoAno As Integer          ' guarda valor desconto Ano
    Dim avenca As String               ' guarda valor avença
    Dim descontoConcelho As Integer     ' guarda valor desconto concelho

    Sub CalcularPreço()
        Dim avenca As String
        Dim total As Integer

        avenca = txtValorBase.Text
        total = CInt(avenca)
        If (chkEscalao.Checked) Then
            descontoEscalao = avenca * 0.1
            total = total - descontoEscalao
            txtTotal.Text = CStr(total)
        Else
            txtTotal.Text = CStr(total)
        End If
        If (chkSocio.Checked) Then
            descontoSocio = avenca * 0.1
            total = total - descontoSocio
            txtTotal.Text = CStr(total)
        Else
            txtTotal.Text = CStr(total)
        End If
        If (chkAgravamento.Checked) Then
            total = total + agravamento
            txtTotal.Text = CStr(total)
        Else
            txtTotal.Text = CStr(total)
        End If
        If (rbtSim.Checked) Then
            descontoDistrito = avenca * 0.01
            total = total - descontoDistrito
            txtTotal.Text = CStr(total)
        Else
            txtTotal.Text = CStr(total)
        End If
        If (cbxConcelho.Text = "Amares") Then
            descontoConcelho = avenca * 0.2
            total = total - descontoConcelho
            txtTotal.Text = CStr(total)
        Else
            txtTotal.Text = CStr(total)
        End If
        If (txtIdade.Text >= "2000") Then
            descontoAno = avenca * 0.1
            total = total - descontoAno
            txtTotal.Text = CStr(total)
        Else
            txtTotal.Text = CStr(total)
        End If


    End Sub

    Private Sub chkEscalao_CheckedChanged(sender As Object, e As EventArgs) Handles chkEscalao.CheckedChanged
        CalcularPreço()
    End Sub

    Private Sub txtValorBase_TextChanged(sender As Object, e As EventArgs) Handles txtValorBase.TextChanged
        If txtValorBase.Text = "" Then
            avenca = "0"
            total = CInt(avenca)
        Else
            avenca = txtValorBase.Text
            total = CInt(avenca)
        End If
        txtTotal.Text = total
    End Sub

    Private Sub chkSocio_CheckedChanged(sender As Object, e As EventArgs) Handles chkSocio.CheckedChanged
        CalcularPreço()
    End Sub

    Private Sub chkAgravamento_CheckedChanged(sender As Object, e As EventArgs) Handles chkAgravamento.CheckedChanged
        CalcularPreço()
    End Sub

    Private Sub rbtSim_CheckedChanged(sender As Object, e As EventArgs) Handles rbtSim.CheckedChanged
        CalcularPreço()
    End Sub

    Private Sub cbxConcelho_SelectedIndexChanged(sender As Object, e As EventArgs) Handles cbxConcelho.SelectedIndexChanged
        CalcularPreço()
    End Sub

    Private Sub txtIdade_TextChanged(sender As Object, e As EventArgs) Handles txtIdade.TextChanged
        If (txtIdade.Text >= "2000") Then
            CalcularPreço()
        End If

    End Sub



    '    Dim total As Integer                ' Variavel guarda total
    '    Dim descontoEscalao As Integer      ' guarda valor desconto escalao
    '    Dim descontoSocio As Integer        ' guarda valor desconto Socio
    '    Dim agravamento As Integer = 12     ' guarda valor agravamento
    '    Dim descontoDistrito As Integer     ' guarda valor desconto distrito
    '    Dim descontoAno As Integer          ' guarda valor desconto Ano
    '    Dim avenca As Integer               ' guarda valor avença
    '    Dim descontoConcelho As Integer     ' guarda valor desconto concelho

    '    avenca = txtValorBase.Text
    '    total = CInt(avenca)

    '    ' Se tiver escalao
    '    If (chkEscalao.Checked) Then
    '        descontoEscalao = avenca * 0.1
    '        total = avenca - descontoEscalao
    '    End If

    '    'Se for socio
    '    If (chkSocio.Checked) Then
    '        descontoSocio = avenca * 0.05
    '        total = total - descontoSocio
    '    End If

    '    'se tiver agravamento
    '    If (chkAgravamento.Checked) Then
    '        total = total + agravamento
    '    End If

    '    'se for maior ou igual a 2000
    '    If (txtIdade.Text >= "2000") Then
    '        descontoAno = avenca * 0.1
    '        total = total - descontoAno
    '    End If

    '    'se for de Amares
    '    If (cbxConcelho.Text = "Amares") Then
    '        descontoConcelho = avenca * 0.2
    '        total = total - descontoConcelho
    '    End If

    '    ' se for do Distrito
    '    If (rbtSim.Checked) Then
    '        descontoDistrito = avenca * 0.01
    '        total = total - descontoDistrito
    '    End If


    '    txtTotal.Text = CStr(total)
    'End Sub

End Class

File: ShrekCS/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace ShrekCS
{
    public partial class Form1 : Form
    {
        private string avenca;
        private int anoNascimento;
        private double total;
        private double descontoEscalao;
        private double descontoSocio;
        private int agravamento = 12;
        private double descontoDistrito;
        private double descontoAno;
        private double descontoConcelho;

        public Form1()
        {
            InitializeComponent();
        }

        void calcular()
        {
            anoNascimento = Convert.ToInt16(txtAnoNascimento.Text);
            avenca = txtValorBase.Text;
            total = Convert.ToInt16(avenca);

            try
            {
                if (txtValorBase.Text != "")
                {
                    if (anoNascimento >= 2000)
                    {
                        descontoAno = Convert.ToInt16(avenca) * 0.1;
                        total = total - descontoAno;
                        txtTotal.Text = total.ToString();
                    }
                    else
                    {
                        txtTotal.Text = total.ToString();
                    }


                    if (chkDescontoEscalao.Checked)
                    {
                        descontoEscalao = Convert.ToInt32(avenca) * 0.1;
                        total = total - descontoEscalao;
                        txtTotal.Text = total.ToString();
                    }
                    else
                    {
                        txtTotal.Text = total.ToString();
                    }

                    if (chkDescontoSocio.Checked)
                    {
                        descontoSocio = Convert.ToInt32(avenca) * 0.1;
                        total = total - descontoSocio;
                        txtTotal.Text = Convert.ToString(total);
                    }
                    else
                    {
                        txtTotal.Text = total.ToString();
                    }

                    if (chkAgravamento.Checked)
                    {
                        total = total + agravamento;
                        txtTotal.Text = total.ToString();
                    }
                    else
                    {
                        txtTotal.Text = total.ToString();
                    }

                    if (rbtSim.Checked)
                    {
                        descontoDistrito = Convert.ToInt16(avenca) * 0.01;
                        total = total - descontoDistrito;
                        txtTotal.Text = total.ToString();
                    }
                    else
                    {
                        txtTotal.Text = total.ToString();
                    }

                    if (cbxConcelhos.Text == "Amares")
                    {
                        descontoConcelho = Convert.ToInt16(avenca) * 0.2;
                        total = total - descontoConcelho;
                        txtTotal.Text = total.ToString();
                    }
                    else
                    {
                        txtTotal.Text = total.ToString();
                    }
                }
            }
            catch (Exception e)
            {
                Console.WriteLine(e);
                throw;
            }

        }

        private void txtValorBase_TextChanged(object sender, EventArgs e)
        {
            try
            {
                if (txtValorBase.Text == "")
                {
                    txtTotal.Text = "0";
                }
                else
                {
                    avenca = txtValorBase.Text;
                    txtTotal.Text = avenca;
                }
            }
            catch (Exception exception)
            {
                Console.WriteLine(exception);
                throw;
            }
            
        }

        private void txtAnoNascimento_TextChanged(object sender, EventArgs e)
        {
            try
            {
                if (txtValorBase.Text != "")
                {
                    calcular();
                }
                else
                {
                    MessageBox.Show("Insira Valor Base Valido");
                }
            }
            catch (Exception exception)
            {
                Console.WriteLine(exception);
                throw;
            }
            
        }

        private void chkDescontoEscalao_CheckedChanged(object sender, EventArgs e)
        {
            calcular();
        }

        private void chkDescontoSocio_CheckedChanged(object sender, EventArgs e)
        {
            calcular();
        }

        private void chkAgravamento_CheckedChanged(object sender, EventArgs e)
        {
            calcular();
        }

        private void rbtSim_CheckedChanged(object sender, EventArgs e)
        {
            calcular();
        }

        private void cbxConcelhos_SelectedIndexChanged(object sender, EventArgs e)
        {
            calcular();
        }
    }
}
